File: week11/Clock.py
from Counter import Counter

class Clock:

    def __init__(self):
        self.second = Counter(name="second")
        self.minute = Counter(name="minute")
        self.hour = Counter(name="hour")
        
    def Tick(self):
        self.second.Increment()
        if self.second.Ticks > 59:
            self.second.Reset()
            self.minute.Increment()

            if self.minute.Ticks > 59:
                self.minute.Reset()
                self.hour.Increment()

                if self.hour.Ticks > 23:
                    self.Reset()

    def Reset(self):
        self.second.Reset()
        self.minute.Reset()
        self.hour.Reset()

    @property
    def Time(self):
        return f"{self.hour.Ticks:02}:{self.minute.Ticks:02}:{self.second.Ticks:02}"

File: week3/CounterTask2/CounterTask2/Clock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CounterTask2
{
    public class Clock
    {
        readonly Counter _seconds = new("seconds");
        readonly Counter _minutes = new("minutes");
        readonly Counter _hours = new("hours");

        public Clock()
        {

        }
        public void Tick()
        {
            _seconds.Increment();
            if (_seconds.Tick > 59)
            {
                _minutes.Increment();
                _seconds.Reset();
                if (_minutes.Tick > 59)
                {
                    _hours.Increment();
                    _minutes.Reset();
                    if (_hours.Tick > 23)
                    {
                        Reset();
                    }
                }
            }
        }

        public void Reset()
        {
            _seconds.Reset();
            _minutes.Reset();
            _hours.Reset();
        }

        public string Time()
        {
            return _hours.Tick.ToString("D2") + ":" + _minutes.Tick.ToString("D2") + ":" + _seconds.Tick.ToString("D2");
        }

    }
}